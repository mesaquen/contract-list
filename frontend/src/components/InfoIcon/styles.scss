.infoIcon--icon {
  color: var(--secondary-yellow);
  cursor: pointer;

  &:hover {
    color: var(--primary-main-titles);
  }
}
