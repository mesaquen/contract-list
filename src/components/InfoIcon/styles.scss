.infoIcon--icon {
    color: var(--primary-yellow);
    cursor: pointer;
    
    &:hover {
      color: var(--primary-main-titles)
    }
}