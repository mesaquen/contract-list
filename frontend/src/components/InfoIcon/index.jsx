import React from 'react'
import PropTypes from 'prop-types'
import { MdInfo } from 'react-icons/md'
import './styles.scss'

const InfoIcon = ({ info, size, ...props }) => {
  return (
    <span title={info} {...props}>
      <MdInfo className="infoIcon--icon" size={size} />
    </span>
  )
}

InfoIcon.propTypes = {
  info: PropTypes.string,
  size: PropTypes.string,
}

InfoIcon.defaultProps = {
  size: '1.5rem',
}

export default InfoIcon
