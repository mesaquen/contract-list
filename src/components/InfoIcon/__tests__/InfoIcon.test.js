import { render, cleanup } from '@testing-library/react'
import InfoIcon from '../index'

beforeEach(cleanup)

describe('InfoIcon', () => {
  test('Matches snapshot', () => {
    const { asFragment } = render(<InfoIcon info="Sample info" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
