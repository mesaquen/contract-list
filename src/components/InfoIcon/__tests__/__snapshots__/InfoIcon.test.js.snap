// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoIcon Matches snapshot 1`] = `
<DocumentFragment>
  <span
    title="Sample info"
  >
    <svg
      class="infoIcon--icon"
      fill="currentColor"
      height="1.5rem"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1.5rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
  </span>
</DocumentFragment>
`;
